test/main-test.js

```diff
@@ -532,6 +537,36 @@ describe('PercyClient', function() {
       nock('https://percy.io')
         .post('/api/v1/snapshots/123/finalize')
         .reply(201, responseData);
+
+      let request = percyClient.finalizeSnapshot(123);
+
+      request
+        .then(response => {
+          assert.equal(response.statusCode, 201);
+          assert.deepEqual(response.body, {success: true});
+          done();
+        })
+        .catch(err => {
+          done(err);
+        });
+    });
+
+    it('finalizes the snapshot with 3 retries', function(done) {
```

Comment:
Needs a test for what happens after 3 retries and it still fails, should be that the status is the same as the last 500 code I think.